app/src/main/java/cm/aptoide/pt/promotions/PromotionsPresenter.java

```diff
@@ -177,7 +202,19 @@ private void handleRetryClick() {
         .filter(event -> event.equals(View.LifecycleEvent.CREATE))
         .flatMap(event -> view.retryClicked()
             .doOnNext(__ -> view.showLoading())
-            .flatMap(__ -> loadPromotionApps())
+            .flatMap(__ -> promotionsManager.getPromotionsModel(promotionId))
+            .observeOn(viewScheduler)
+            .doOnNext(
```

Comment:
Same logic as in the getPromotionApps chain, maybe we could extract this?
Also, same as the above regarding the appcoins value.

Reply:
fixed